### PR TITLE
Vestax VCI-100MKII: map "Rate by quantized BPM" to [shift + pitch fader]

### PR DESCRIPTION
**Pull request by sohet**
- Vestax VCI-100MKII: map "Rate by quantized BPM" to [shift + pitch fader]
  May be a nice companion of pitch fader for those who want exact BPM but don't want sync mode. It enables a user to set BPM of integer values. A drift of BPM, caused by mechanical weakness, may happen while playing for a long time after setting rate by quantized BPM. So I add an automatic lock: locked at any move of [shift + pitch fader] and unlocked at the move of [pitch fader] greater than 1 MIDI value of MSB (just as the inverse of soft-takeover).
- Vestax VCI-100MKII: clean up the code of exchanging fx display by shift
  Replace VCI102.shift/VCI102.setShift by VCI102.fxButton/VCI102.setFxButton. No change in the behavior.

### CODE REVIEW

**Comment by illuusio** [comment]
Are these naming changes somehow more descriptive than older ones? I Have to re-read code to fully understand what this really does. So you can fine tune BPM in long running session?

**Comment by sohet** [comment]
Intended behaviors are:

* pitch fader (mapped to VCI102.rateMSB and VCI102.rateLSB)

    Change rate within "rateRange" continuously as usual.

* shift + pitch fader (mapped to VCI102.rateQuantizedMSB and VCI102.rateQuantizedLSB)

    Change rate within "rateRange" by quantized BPM (128, 129, 130, ..., limited to exact integer values, not for "fine tuning").

After the move of [shift + pitch fader], any changes of rate by the move of [pitch fader] within 1 MIDI value (of MSB) are inhibited. This lock is against rate drift caused by mechanical weakness (I don't know whether it is necessary for other controllers). If the move exceeds 1 MIDI value, changes of rate by [pitch fader] are permitted again. For the move of 1 MIDI value is too small not to be exceeded by a user (at least in the case of this controller), he can change rate again by [pitch fader] without any special intentions to unlock it.

**Comment by illuusio** [comment]
Ok it's mechanical thing. Variable names are more informative for me :+1: . I can Imagine many other MIDI devices are suffering this kind of problems. So this one adds more steps. Code it self look very straight forward and clean but have you tried to play set with these tweaks. I'm just fearing if you add more steps and lock are they opening as expecting?

**Comment by sohet** [comment]
Thank you for your suggestion. I played more than an hour using this script and found no difficulty about unlocking. Soft-takeover seems almost always tolerate enough to permit "rate" changes after uncontinuous "bpm" changes when "rateRange" is the same. But I also noticed that at a rare case it surely privents the change (ex. at file_bpm=140 and rateRange=0.08, changing rate by quantized BPM to 135, then an upward change of rate is prevented). So I include temporary cancellation of soft-takeover to the unlocking procedure.

File: res/controllers/Vestax-VCI-100MKII-scripts.js [hunk 1]
```diff
@@ -118,14 +127,44 @@ VCI102.jog = function(ch, midino, value, status, group) {
     }
 };
 
-VCI102.rate = [0, 0, 0, 0];
+VCI102.rate7bit = [64, 64, 64, 64];
+
```
Comment (illuusio) [inline res/controllers/Vestax-VCI-100MKII-scripts.js]: Why did 64 is chosen. Ok 7Bits but at least some explanation would we good.

File: res/controllers/Vestax-VCI-100MKII-scripts.js [hunk 2]
```diff
@@ -118,14 +127,44 @@ VCI102.jog = function(ch, midino, value, status, group) {
...
+VCI102.rateEnable = [true, true, true, true];
 
 VCI102.rateMSB = function(ch, midino, value, status, group) {
-    VCI102.rate[ch] = (64 - value) * 128;
```
Comment (illuusio) [inline res/controllers/Vestax-VCI-100MKII-scripts.js]: Please add some comments to someone who looks this first time what this should do and why.

File: res/controllers/Vestax-VCI-100MKII-scripts.js [hunk 3]
```diff
@@ -118,14 +127,44 @@ VCI102.jog = function(ch, midino, value, status, group) {
...
+    }
+    VCI102.rate7bit[ch] = value;
+};
+
+VCI102.rate = function(ch, lsb) {
+    return ((64 - VCI102.rate7bit[ch]) * 128 - lsb) / 8192;
 };
 
 VCI102.rateLSB = function(ch, midino, value, status, group) {
-    engine.setValue(group, "rate", (VCI102.rate[ch] - value) / 8192);
```
Comment (illuusio) [inline res/controllers/Vestax-VCI-100MKII-scripts.js]: Again some comments would clarify things

File: res/controllers/Vestax-VCI-100MKII-scripts.js (hunk 1 above)
Comment (sohet) [inline res/controllers/Vestax-VCI-100MKII-scripts.js]: This variable is for raw MIDI values of 7bit-MSB. For the default will never be used, anything do. But the choice of 64 (= 0x40, center) is natural for a tentative value of pitch fader, I think.

Comment (illuusio) [inline res/controllers/Vestax-VCI-100MKII-scripts.js]: Yes I see why it's taken but it should be also tell to all mankind.

**Comment by sohet** [comment]
I believe ready to merge.

**Comment by illuusio** [comment]
Sorry bit lack of time but Merge

**Comment by ywwg** [comment]
sohet thans, and illuusio thanks for review!

**Pull request merged**